packages/application-shell/src/test-utils/test-utils.js

```diff
@@ -60,6 +62,16 @@ const defaultPermissions = { canManageProject: true };
 const mergeOptional = (defaultValue, value) =>
   value === null ? undefined : { ...defaultValue, ...value };
 
+const ApolloProvider = props =>
+  props.mocks ? (
+    <ApolloMockProvider {...props} />
+  ) : (
+    <ApolloActualProvider client={createApolloClient()}>
+      {props.children}
+    </ApolloActualProvider>
+  );
+ApolloProvider.displayName = 'ApolloProvider';
```

Comment:
@pa3 How do you feel about creating a separate `render` function to keep the experimental stuff out of our actual `render` fn used in tests?

Reply:
That can be done, but I am also happy with the semantics: "you didn't gave us mocks, you probably don't want to mock Apollo".

Reply:
From what I understand you're still experimenting with this new approach. Adding it to the "official" `render` function makes it seem as-if it is a recommended way already. That's why I would prefer a separate function for it as we might want to change/remove it after the experimentation.

Reply:
@dferber90 Updated this PR to have additional `experimentalRender` function.